app/process/watcher/crypto-transfer/crypto-transfer.go

```diff
@@ -1,3 +1,88 @@
-package crypto_transfer
+package cryptotransfer
 
-// crypto transfer watcher
+import (
+	"github.com/hashgraph/hedera-sdk-go"
+	hederaClient "github.com/limechain/hedera-eth-bridge-validator/app/clients/hedera"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/repositories"
+	cryptotransfermessage "github.com/limechain/hedera-eth-bridge-validator/app/process/model/crypto-transfer-message"
+	"github.com/limechain/hedera-eth-bridge-validator/app/process/watcher/publisher"
+	"github.com/limechain/hedera-watcher-sdk/queue"
+	log "github.com/sirupsen/logrus"
+	"time"
+)
+
+type CryptoTransferWatcher struct {
+	client           *hederaClient.HederaClient
+	accountID        hedera.AccountID
+	typeMessage      string
+	pollingInterval  time.Duration
+	statusRepository repositories.StatusRepository
+	maxRetries       int
+}
+
+func NewCryptoTransferWatcher(client *hederaClient.HederaClient, accountID hedera.AccountID, pollingInterval time.Duration, repository repositories.StatusRepository, maxRetries int) *CryptoTransferWatcher {
+	return &CryptoTransferWatcher{
+		client:           client,
+		accountID:        accountID,
+		typeMessage:      "HCS_CRYPTO_TRANSFER",
+		pollingInterval:  pollingInterval,
+		statusRepository: repository,
+		maxRetries:       maxRetries,
+	}
+}
+
+func (ctw CryptoTransferWatcher) Watch(q *queue.Queue) {
+	go ctw.beginWatching(q)
+}
+
+func (ctw CryptoTransferWatcher) beginWatching(q *queue.Queue) {
+	if !ctw.client.AccountExists(ctw.accountID) {
+		log.Errorf("Error incoming: Could not start monitoring account [%s]\n", ctw.accountID.String())
+	}
+	log.Infof("Started Crypto Transfer Watcher for account [%s]\n", ctw.accountID)
+	milestoneTimestamp := ctw.statusRepository.GetLastFetchedTimestamp(ctw.accountID)
+	for {
+		transactions, e := ctw.client.GetAccountTransactionsAfterDate(ctw.accountID, milestoneTimestamp)
+		if e != nil {
+			log.Errorf("Error incoming: Suddenly stopped monitoring account [%s]\n", ctw.accountID.String())
+			log.Errorln(e)
+			ctw.restart(q)
+			return
+		}
+
+		if len(transactions.Transactions) > 0 {
+			for _, tx := range transactions.Transactions {
+				log.Infof("[%s] - New transaction on account [%s] - Tx Hash: [%s]\n",
+					tx.ConsensusTimestamp,
+					ctw.accountID.String(),
+					tx.TransactionHash)
+
+				information := cryptotransfermessage.CryptoTransferMessage{
+					TxMemo: tx.MemoBase64,
+					Sender: tx.Transfers[len(tx.Transfers)-2].Account,
+					Amount: tx.Transfers[len(tx.Transfers)-1].Amount,
```

Comment:
I don't think that the indices -1 and -2 should be hardcoded here. What if there are 10 transfers in the transaction? will this hold?

Reply:
As far as I researched, when a crypto-transfer happens, the Receiver is on the second-to-last position and the Sender is on the last one, no matter how many transfers occur.

Reply:
A new way to retrieve the data from the transfers will be implemented